File: level1/ex46.js
function solution(n) {
  // 약수 배열 구하는 함수
  function getDivisor(number) {
      const divisor = [];
      
      for (let i = 1; i <= number; i++) {
          if (number % i === 0) {
              divisor.push(i);
          }
      }
      
      return divisor;
  }
  
  const divisor = getDivisor(n);
  
  return divisor.length > 0 ? divisor.reduce((a, b) => a + b) : 0
}

File: level1/ex46.py
def solution(n):
    # n이 0인 경우
    if n == 0:
        return 0
    
    # 약수 리스트
    divisor = [i for i in range(1, n + 1) if n % i == 0]
    # 합
    divisorSum = sum(divisor)
    
    return divisorSum